src/ast/ctx.rs

```diff
@@ -96,45 +106,89 @@ type GenericCacheMap = IndexMap<String, Arc<RefCell<IndexMap<String, Arc<RefCell
 pub type GenericCache = Arc<RefCell<GenericCacheMap>>;
 
 /// # Ctx
-/// Context for code generation
+/// Context for code generation and LSP features
 pub struct Ctx<'a> {
     pub generic_types: FxHashMap<String, Arc<RefCell<PLType>>>,
-    pub need_highlight: Arc<RefCell<usize>>,
     pub plmod: Mod,
-    pub father: Option<&'a Ctx<'a>>, // father context, for symbol lookup
+
+    pub parent: Option<&'a Ctx<'a>>, // parent context, for symbol lookup
     pub root: Option<&'a Ctx<'a>>,   // root context, for symbol lookup
-    pub function: Option<ValueHandle>, // current function
-    pub init_func: Option<ValueHandle>, //init function,call first in main
-    pub block: Option<BlockHandle>,  // current block
-    pub continue_block: Option<BlockHandle>, // the block to jump when continue
-    pub break_block: Option<BlockHandle>, // the block to jump to when break
-    pub return_block: Option<(BlockHandle, Option<ValueHandle>)>, // the block to jump to when return and value
-    pub errs: &'a RefCell<FxHashSet<PLDiag>>,                     // diagnostic list
-    pub edit_pos: Option<Pos>,                                    // lsp params
-    pub table: FxHashMap<String, PLSymbolData>,                   // variable table
-    pub config: Config,                                           // config
+
+    /// LSP argument, because the high light among multiple tokens require an order,
+    /// hence, we stores a value inside it to refer whether we need to generate the code or not.
+    /// highlight is generated only if the number is 0
+    pub need_highlight: Arc<RefCell<usize>>,
+
+    /// LLVM argument: the index to mark a function element inside builder
```

Comment:
The point of `Handles` is to separate lowlevel codegen logic and ast lowering. Hence it is not bind to LLVM. Although currently LLVM is the only working code gen backend, we may have more in future.